ssl/t1_lib.c

```diff
@@ -577,19 +578,15 @@ void tls1_get_formatlist(SSL *s, const unsigned char **pformats,
 /* Check a key is compatible with compression extension */
 static int tls1_check_pkey_comp(SSL *s, EVP_PKEY *pkey)
```

Comment:
This function is inextricably linked with the point encoding/decoding. When we push the point encoding/decoding provider side then we will also have to push the logic about what point compression is supported provider side too, i.e. we should ask the provider what point compression is supported for any particular key - rather than trying to infer if from the field type. That works for own stuff because we know we support point compression - but we can't necessarily assume the same for provider owned keys. For now though we are explicitly downgrading EC keys prior to point encoding/decoding in order to use the ameth implementation. It seems we could do the same trick here and then just do it the old way - until such time as the whole thing goes provider side.

Reply:
Okie, done